### ISSUE TITLE
Exception thrown in Flutter Web randomly when text field is used

### ISSUE THREAD

**Issue by cupidchan** [flutter/flutter#119516]
## Steps to Reproduce
1. Create a brand new Flutter project using Flutter CLI and then use the sample at https://github.com/firebase/flutterfire/tree/master/packages/firebase_ui_auth/example.
2. Try to register a user and exception thrown.

**Expected results:** 
No exception should be thrown.

**Actual results:** 
Please see the video below. It shows 3 issues and the first one is for this bug. Moreover, it's not always reproducible and looks like some events will trigger it more often. 
 https://user-images.githubusercontent.com/12452100/214127630-b78d52cd-4b71-48c6-ba5a-e964a7d1f881.mov

<details>
<summary>Code sample</summary>
Here is the repo I created to reproduce this issue: https://github.com/cupidchan/firebase_ui_auth_issue10279 
</details>

<details>
  <summary>Logs</summary>

```
errors.dart:266 Uncaught Error: Assertion failed: file:///opt/homebrew/Caskroom/flutter/3.3.6/flutter/packages/flutter/lib/src/services/hardware_keyboard.dart:435:16
_pressedKeys[event.physicalKey] == event.logicalKey
"A KeyUpEvent is dispatched, but the state shows that the physical key is pressed on a different logical key. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyUpEvent#c185a(physicalKey: PhysicalKeyboardKey#7000c(usbHidUsage: \"0x0007000c\", debugName: \"Key I\"), logicalKey: LogicalKeyboardKey#00069(keyId: \"0x00000069\", keyLabel: \"I\", debugName: \"Key I\"), character: null, timeStamp: 0:00:09.265000, synthesized) and the recorded logical key LogicalKeyboardKey#70b1f(keyId: \"0x17c007000c\", keyLabel: \"\", debugName: \"Key with ID 0x017c007000c\")"
    at Object.throw_ [as throw] (errors.dart:266:49)
    at Object.assertFailed (errors.dart:29:3)
    at hardware_keyboard.dart:435:56
    at [_assertEventIsRegular] (hardware_keyboard.dart:442:14)
    at hardware_keyboard.HardwareKeyboard.new.handleKeyEvent (hardware_keyboard.dart:535:5)
    at hardware_keyboard.KeyEventManager.new.handleKeyData (hardware_keyboard.dart:808:29)
    at platform_dispatcher.dart:273:33
    at Object.invoke (platform_dispatcher.dart:1168:13)
    at _engine.EnginePlatformDispatcher.new.invokeOnKeyData (platform_dispatcher.dart:272:7)
    at [_onKeyData] (keyboard_binding.dart:134:39)
    at keyboard_binding.dart:539:36
    at [_handleEvent] (keyboard_binding.dart:416:11)
    at _engine.KeyboardConverter.new.handleEvent (keyboard_binding.dart:542:7)
    at keyboard_binding.dart:141:25
    at Object._checkAndCall (operations.dart:334:14)
    at Object.dcall (operations.dart:339:39)
    at ret (js_patch.dart:413:11)
    at loggedHandler (keyboard_binding.dart:112:23)
    at Object._checkAndCall (operations.dart:334:14)
    at Object.dcall (operations.dart:339:39)
    at ret (js_patch.dart:413:11)

```
</details>

```
Analyzing firebase_ui_auth_issue10279...                                
No issues found! (ran in 1.1s)
```

```
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.1 22C65 darwin-arm, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.74.3)
[✓] Connected device (2 available)
[✓] HTTP Host Availability
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @cupidchan 
Looking at the error log, it seems similar to already open issue https://github.com/flutter/flutter/issues/100455 in which you can follow-up for further updates. Closing this as a duplicate. If you disagree, write in comments and I'll reopen it.

**Comment by bleroux** [comment]
@cupidchan
I will discussed with @darshankawar to see if we should reopen this issue. Based on my testings #100455 main code sample does not throws in 3.7.0 (but there is still something wrong because it does not print when expected). So it might be a different issue, we need more testings to confirm that.

In the meantime, can you create a more minimal code sample? I tried to launch your sample app but it failed with a Firebase error.

**Comment by bleroux** [comment]
@cupidchan
Thanks for updating your sample app, I'm able to launch the app but I can't reproduce the error.

> Please note that it error seems to happen only in Debug, but ok in Run.

This is expected as this error is a check that it is only done in debug mode (an assert in Flutter code). It is a somewhat generic warning that something related to a key event went wrong. This is usually not a problem in production but it could be. There are so much keyboards layout, languages, systems specifics settings, involved that this warning is helpful to improve Flutter keyboard management.

It would be very helpful to see if you can repro this error with a very basic Flutter app with only one TextField such as:

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Container(
          alignment: Alignment.center,
          padding: const EdgeInsets.all(16),
          child: const TextField(),
        ),
      ),
    );
  }
}
```
</details>

Another interesting feedback would be to know if you get this error always doing the same key sequences.
From your error log, it was related to the 'I' key (`PhysicalKeyboardKey#7000c(usbHidUsage: \"0x0007000c\", debugName: \"Key I\"`). Would be great, if you can to repro it several times and see if the error message is always related to same key.

**Comment by cupidchan** [comment]
Thanks @bleroux! You gave me the hint to reproduce this consistently!

1. Launch the app `flutter run`
2. Type "1@email.com" in the user name text field. 
3. The moment the "i" is type, it throws the exception.

There may be other key sequence triggering this issue, but this one for sure I can trigger the exception consistently on my side. Can you please try?

**Comment by darshankawar** [comment]
Thanks @cupidchan for the update and @bleroux for sharing your insights.

Is this only occuring using physical keyboard (magic) ?

I'll verify this at my end and share my results in here.

Reopening for better tracking.

**Comment by cupidchan** [comment]
@darshankawar, since my app is web app, i always use physical key for testing and found this issue.

**Comment by darshankawar** [comment]
I cloned the original repo shared and ran it on web, tried inputing `1@email.com` using physical keyboard (apple magic wireless keyboard connected via bluetooth), but didn't get the reported exception.

<img width="1148" alt="Screenshot 2023-02-03 at 11 11 38 AM" src="https://user-images.githubusercontent.com/67046386/216524796-57e2eefa-6731-4767-b8a9-157ad1b5e148.png">

I also tried with a minimal reproducible code sample mentioned in above comments but didn't see the exception being triggered.

I verified this on mac + web + magic keyboard, also with laptop keyboard as well on latest stable (3.7.1) as well as master.

So, this could be an edge case with specific config as mentioned by @bruno in above comment.
